### PR TITLE
docs: add build instructions

### PR DESCRIPTION
**Pull request by glennsl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,30 @@ TODO, read the [mli](./lib/piaf.mli) file for now.
...
+
+There's two ways to get a development environemnt up and running. If you have (or don't mind getting) [`nix`](https://nixos.org/nix/manual/) installed, the repository includes scripts to set up a sandbox. Otherwise you can use [`opam`](https://opam.ocaml.org/) to install the necessary dependencies globally.
+
+### Option 1) Setting up the sandbox
+
+Assuming [`nix` has been installed](https://nix.dev/tutorials/install-nix.html) and set up, run `nix-shell --pure` in the repository root. Once it's done building, you should have the development environment set up!
+
+### Option 2) Setting up opam
+
+For this approach you'll need to [install `opam`](https://opam.ocaml.org/doc/Install.html)  and set it up with a switch using `ocaml` >= 4.06. Once that's done, run `opam pin . --deps-only` to install the dependencies.
```
Comment (anmonteiro) [inline README.md:36]: I think this actually only works since OCaml 4.08

File: README.md [hunk 2]
```diff
@@ -23,6 +23,30 @@ TODO, read the [mli](./lib/piaf.mli) file for now.
...
+
+Note that this installs the dependencies globally, and that the development environment is dependent on the switch used.
+
+### Building
+
+Run `dune build` to build, `dune build --watch` to run a watcher dameon that will build incremenetally.
```
Comment (anmonteiro) [inline README.md:42]:
```suggestion
Run `dune build` to build, `dune build --watch` to run a watcher dameon that will build incrementally.
```

File: README.md [hunk 3]
```diff
@@ -23,6 +23,30 @@ TODO, read the [mli](./lib/piaf.mli) file for now.
...
+
+### Running examples
+
+Run `dune exec examples/docs/readme.exe` to run the simple example above.
+
+Run `dune exec bin/carl.exe` to run the `carl`, the `curl`-like example.
```
Comment (anmonteiro) [inline README.md:48]:
```suggestion
Run `dune exec bin/carl.exe` to run `carl`, the `curl`-like example.
```

File: README.md (hunk 1 above)
Comment (glennsl) [reply README.md:36]: Ah, I just took that from the opam file. I can test to make sure though.

File: README.md (hunk 3 above)
Comment (glennsl) [reply README.md:48]: :facepalm:

File: README.md (hunk 1 above)
Comment (anmonteiro) [reply README.md:36]: maybe the opam file should also be updated. I haven't looked at it in a while

Comment (glennsl) [reply README.md:36]: Oh, actually I can't test, because I have gcc 10 and nothing below 4.09.1 works for me...

Comment (glennsl) [reply README.md:36]: Let's say 4.08 just to be safe though

Comment (glennsl) [reply README.md:36]:
```suggestion
For this approach you'll need to [install `opam`](https://opam.ocaml.org/doc/Install.html)  and set it up with a switch using `ocaml` >= 4.08. Once that's done, run `opam pin . --deps-only` to install the dependencies.
```

File: README.md [hunk 4]
```diff
@@ -23,9 +23,32 @@ TODO, read the [mli](./lib/piaf.mli) file for now.
...
+
+### Option 2) Setting up opam
+
+For this approach you'll need to [install `opam`](https://opam.ocaml.org/doc/Install.html)  and set it up with a switch using `ocaml` >= 4.08. Once that's done, run `opam pin . --deps-only` to install the dependencies.
+
+Note that this installs the dependencies globally, and that the development environment is dependent on the switch used.
+
+### Building
+
+Run `dune build` to build, `dune build --watch` to run a watcher dameon that will build incrementally.
```
Comment (anmonteiro) [inline README.md:42]:
```suggestion
Run `dune build` to build, `dune build --watch` to run a watcher daemon that will build incrementally.
```

**Pull request merged**